src/Compilers/CSharp/Portable/Syntax/SyntaxTokenParser.cs

```diff
@@ -54,6 +56,36 @@ public Result ParseNextToken()
         return new Result(new SyntaxToken(parent: null, token, startingPosition, index: 0), startingDirectiveStack);
     }
 
+    /// <summary>
+    /// Parse the leading trivia of the next token from the input at the current position. This will advance the internal position of the
+    /// token parser to the end of the leading trivia of the next token. The returned result will have a token with <see cref="CSharpExtensions.Kind(SyntaxToken)"/>
+    /// of <see cref="SyntaxKind.None"/>, <see cref="SyntaxToken.IsMissing"/> set to <see langword="true"/>, and a parent of <see langword="null"/>. The
+    /// parsed trivia will be set as the <see cref="SyntaxToken.LeadingTrivia"/> of the token.
+    /// </summary>
+    public Result ParseLeadingTrivia()
+    {
+        var startingDirectiveStack = _lexer.Directives;
+        var startingPosition = _lexer.TextWindow.Position;
+        var leadingTrivia = _lexer.LexSyntaxLeadingTrivia();
+        var containingToken = InternalSyntax.SyntaxFactory.MissingToken(leading: leadingTrivia.Node, SyntaxKind.None, trailing: null);
+        return new Result(new SyntaxToken(parent: null, containingToken, startingPosition, index: 0), startingDirectiveStack);
+    }
+
+    /// <summary>
+    /// Parse syntax trivia from the current position, according the rules of trailing syntax trivia. This will advance the internal position of the
```

Comment:
```suggestion
    /// Parse syntax trivia from the current position, according to the rules of trailing syntax trivia. This will advance the internal position of the
```